sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -367,6 +392,10 @@ RT::PiProgram ProgramManager::loadProgram(OSModuleHandle M,
   }
   // Load the selected image
   const RT::PiContext &Ctx = getRawSyclObjImpl(Context)->getHandleRef();
+  if ((Format == PI_DEVICE_BINARY_TYPE_SPIRV || Format == PI_DEVICE_BINARY_TYPE_LLVMIR_BITCODE) &&
+      !is_compiler_available(Ctx)) {
```

Comment:
If you pass SYCL context here instead of PI context, `is_compiler_available` can be written like this:
```c++
static bool is_compiler_available(const context &C) {
   for (const auto &D : C.get_devices()) {
      if (D.get_info<info::device::is_compiler_available>()) {
        return true;
      }
    }
  return false;
}
```

But I would replace `is_compiler_available` with `is_program_il_supported`.